### PR TITLE
Fix Issue#566 and unblock PR#1201.

### PR DESCRIPTION
**Pull request by pallavit**
Microsoft.Win32.Registry test made incorrect assumptions about the Registry's unnamed value's default value. Updated the tests to remove the assumption.

### CODE REVIEW

File: src/Microsoft.Win32.Registry/tests/Helpers.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.Win32.RegistryTests
+{
+    internal static class Helpers
+    {
+        [DllImport("api-ms-win-core-registry-l2-1-0.dll", CharSet = CharSet.Unicode, EntryPoint = "RegSetValueW", SetLastError = true)]
+        private static extern int RegSetValue(SafeRegistryHandle handle, string value, int regType, String sb, int sizeIgnored);
+
+        internal const string _defaultValue = "default";
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Helpers.cs]: Nit: constants should be named in PascalCase

File: src/Microsoft.Win32.Registry/tests/Helpers.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.Win32.RegistryTests
+{
+    internal static class Helpers
+    {
+        [DllImport("api-ms-win-core-registry-l2-1-0.dll", CharSet = CharSet.Unicode, EntryPoint = "RegSetValueW", SetLastError = true)]
+        private static extern int RegSetValue(SafeRegistryHandle handle, string value, int regType, String sb, int sizeIgnored);
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Helpers.cs]: Nit: `string`

File: src/Microsoft.Win32.Registry/tests/Helpers.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        internal const string _defaultValue = "default";
+
+        internal static bool SetDefaultValue(this RegistryKey key)
+        {
+            const int REG_SZ = 1;
+            return RegSetValue(key.Handle, null, REG_SZ, _defaultValue, 0) == 0;
+        }
+
+
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Helpers.cs]: Nit: one extra new line)

File: src/Microsoft.Win32.Registry/tests/Helpers.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        internal static bool IsDefaultValueSet(this RegistryKey key)
+        {
+            const int ERROR_FILE_NOT_FOUND = 2;
+
+            byte[] b = new byte[4];
+            int size = 4;
+            return RegQueryValueEx(key.Handle, null, null, IntPtr.Zero, b, ref size) != ERROR_FILE_NOT_FOUND;
+        }
+    }
+}
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Helpers.cs]: Nit: missing new line)

File: src/Microsoft.Win32.Registry/tests/Microsoft.Win32.Registry.Tests.csproj [hunk 5]
```diff
@@ -45,6 +45,7 @@
     <Compile Include="Registry\Registry_Getvalue_str_str_obj.cs" />
     <Compile Include="Registry\Registry_SetValue_str_str_obj.cs" />
     <Compile Include="Registry\Registry_SetValue_str_str_obj_valuekind.cs" />
+	<Compile Include="Helpers.cs" />
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Microsoft.Win32.Registry.Tests.csproj]: Nit: tab indent?

**Comment by Maxwe11** [comment]
A couple of nit comments but otherwise LGTM and fixes #566

**Comment by pallavit** [comment]
@tarekgh Could you please take a look at this.

**Comment by tarekgh** [comment]
LGTM too.

**Comment by pallavit** [comment]
@dotnet-bot test this please.

**Comment by Maxwe11** [comment]
@dotnet-bot test this please.

**Comment by pallavit** [comment]
@Maxwe11 Thanks for the link :). Merging the change :).

**Pull request merged**